### PR TITLE
fix(pydantic): turn on pydantic smart union

### PR DESCRIPTION
**Pull request by numb3r3**
attempt to address https://github.com/jina-ai/docarray/issues/72

### CODE REVIEW

File: docarray/document/pydantic_model.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ class PydanticDocument(BaseModel):
     text: Optional[str]
     weight: Optional[float]
     uri: Optional[str]
-    tags: Optional[Dict[str, '_StructValueType']]
+    tags: Optional[Dict[str, Any]]
```
Comment (hanxiao) [inline docarray/document/pydantic_model.py:41]: This however makes `tags` unschema

Comment (hanxiao) [reply docarray/document/pydantic_model.py:41]: @JoanFM

Comment (JoanFM) [reply docarray/document/pydantic_model.py:41]: I think is right having tags unschema, it is the free part of Document

Comment (hanxiao) [reply docarray/document/pydantic_model.py]: but the "free-part" still have to be  a valid JSON

Comment (hanxiao) [reply docarray/document/pydantic_model.py]: the right way is to turn on pydantic smart union, as in https://pydantic-docs.helpmanual.io/usage/model_config/#smart-union

**Comment by hanxiao** [comment]
the original title of the PR is misleading, the default behavior is not "smart union" according to https://pydantic-docs.helpmanual.io/usage/model_config/#smart-union

This PR in fact turns on the "smart union".

**Review by hanxiao — APPROVED** [review]

**Pull request merged**